.teamcity/src/main/kotlin/promotion/PublishNightlyDocumentation.kt

```diff
@@ -21,33 +21,16 @@ import jetbrains.buildServer.configs.kotlin.triggers.schedule
 import model.StageName
 import vcsroots.gradlePromotionBranches
 
+const val NIGHTLY_DOCUMENTATION_PROMOTION_ID = "Promotion_NightlyDocumentation"
 class PublishNightlyDocumentation(branch: VersionedSettingsBranch) : PublishGradleDistributionFullBuild(
     promotedBranch = branch.branchName,
     promoteTask = "publishBranchDocs",
     triggerName = StageName.PULL_REQUEST_FEEDBACK.uuid,
     vcsRootId = gradlePromotionBranches
 ) {
     init {
-        id("Promotion_NightlyDocumentation")
+        id(NIGHTLY_DOCUMENTATION_PROMOTION_ID)
         name = "Nightly Documentation"
         description = "Promotes the latest successful documentation changes on '${branch.branchName}' from Ready for Nightly as a new nightly documentation snapshot"
-
-        triggers {
-            branch.nightlyPromotionTriggerHour?.let { triggerHour ->
-                schedule {
-                    schedulingPolicy = daily {
-                        this.hour = triggerHour
-                    }
-                    triggerBuild = always()
-                    withPendingChangesOnly = true
-                    enabled = branch.enableVcsTriggers
-                    // https://www.jetbrains.com/help/teamcity/2022.04/configuring-schedule-triggers.html#general-syntax-1
-                    // We want it to be triggered only when there're pending changes in the specific vcs root, i.e. GradleMaster/GradleRelease
-                    triggerRules = "+:root=${VersionedSettingsBranch.fromDslContext().vcsRootId()}:."
```

Comment:
Is there a way to express the dependency but keep this job "optional"?
That is it would only run if there were pending changes.